File: extract_excel_data.ts
import * as XLSX from 'xlsx';
import * as fs from 'fs';

// constructions.xlsx 읽기
const constructionsWorkbook = XLSX.readFile('constructions.xlsx');
const constructionsSheet = constructionsWorkbook.Sheets[constructionsWorkbook.SheetNames[0]];
const constructionsData = XLSX.utils.sheet_to_json(constructionsSheet);

const constructionNames = [...new Set(constructionsData.map((row: any) => row['업체명']))];

console.log(`총 건설사 수: ${constructionNames.length}`);
console.log('\n건설사 목록:');
constructionNames.forEach((name, index) => {
  console.log(`${index + 1}. ${name}`);
});

// items.xlsx 읽기
const itemsWorkbook = XLSX.readFile('items.xlsx');
const itemsSheet = itemsWorkbook.Sheets[itemsWorkbook.SheetNames[0]];
const itemsData = XLSX.utils.sheet_to_json(itemsSheet);

console.log(`\n총 품목 수: ${itemsData.length}`);
console.log('\n품목 목록:');
itemsData.forEach((item: any) => {
  console.log(`- ${item.item_id}: ${item.item_name}`);
});

// SQL INSERT 문 생성
console.log('\n\n=== SQL INSERT 문 ===\n');

// constructions 테이블 INSERT
console.log('-- constructions 테이블 데이터 삽입');
console.log('INSERT INTO constructions (company_name) VALUES');
const constructionValues = constructionNames.map(name => {
  const escaped = name.replace(/'/g, "''");
  return `('${escaped}')`;
});
console.log(constructionValues.join(',\n'));
console.log('ON CONFLICT (company_name) DO NOTHING;');

// items 테이블 INSERT
console.log('\n-- items 테이블 데이터 삽입');
console.log('INSERT INTO items (item_id, item_name) VALUES');
const itemValues = itemsData.map((item: any) => {
  const escapedId = item.item_id.replace(/'/g, "''");
  const escapedName = item.item_name.replace(/'/g, "''");
  return `('${escapedId}', '${escapedName}')`;
});
console.log(itemValues.join(',\n'));
console.log('ON CONFLICT (item_id) DO NOTHING;');

// JSON 형식으로도 저장
const data = {
  constructions: constructionNames,
  items: itemsData
};

fs.writeFileSync('excel_data.json', JSON.stringify(data, null, 2), 'utf-8');
console.log('\n\nJSON 파일로도 저장되었습니다: excel_data.json');

// SQL 파일로도 저장
const sqlContent = `-- constructions 테이블 데이터 삽입
INSERT INTO constructions (company_name) VALUES
${constructionValues.join(',\n')}
ON CONFLICT (company_name) DO NOTHING;

-- items 테이블 데이터 삽입
INSERT INTO items (item_id, item_name) VALUES
${itemValues.join(',\n')}
ON CONFLICT (item_id) DO NOTHING;`;

fs.writeFileSync('initial_data.sql', sqlContent, 'utf-8');
console.log('SQL 파일로도 저장되었습니다: initial_data.sql');

File: extract_excel_data.py
#!/usr/bin/env python3
import pandas as pd
import json

# constructions.xlsx 읽기
constructions_df = pd.read_excel('constructions.xlsx')
constructions = constructions_df['업체명'].unique().tolist()

print(f"총 건설사 수: {len(constructions)}")
print("\n건설사 목록:")
for i, company in enumerate(constructions, 1):
    print(f"{i}. {company}")

# items.xlsx 읽기
items_df = pd.read_excel('items.xlsx')
items = []
for _, row in items_df.iterrows():
    items.append({
        'item_id': row['item_id'],
        'item_name': row['item_name']
    })

print(f"\n\n총 품목 수: {len(items)}")
print("\n품목 목록:")
for item in items:
    print(f"- {item['item_id']}: {item['item_name']}")

# SQL INSERT 문 생성
print("\n\n=== SQL INSERT 문 ===\n")

# constructions 테이블 INSERT
print("-- constructions 테이블 데이터 삽입")
print("INSERT INTO constructions (company_name) VALUES")
construction_values = []
for company in constructions:
    # SQL 이스케이프 처리
    escaped_company = company.replace("'", "''")
    construction_values.append(f"('{escaped_company}')")

print(",\n".join(construction_values))
print("ON CONFLICT (company_name) DO NOTHING;")

print("\n-- items 테이블 데이터 삽입")
print("INSERT INTO items (item_id, item_name) VALUES")
item_values = []
for item in items:
    item_id = item['item_id'].replace("'", "''")
    item_name = item['item_name'].replace("'", "''")
    item_values.append(f"('{item_id}', '{item_name}')")

print(",\n".join(item_values))
print("ON CONFLICT (item_id) DO NOTHING;")

# JSON 형식으로도 저장
data = {
    'constructions': constructions,
    'items': items
}

with open('excel_data.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)

print("\n\nJSON 파일로도 저장되었습니다: excel_data.json")